feature(Implement link voucher to stock movement)

lomamech opened the pull request:
- Implement link stock movement receipt from Vouchers for all Stock
  movement

closes #5504

lomamech committed:
feature(Implement link voucher to stock movement)
- Implement link stock movement receipt from Vouchers for all Stock
  movement

closes #5504

lomamech committed:
Improvement: remove the join and show the link only if the transaction type is for stock movement

jniles commented on client/src/modules/vouchers/templates/action.cell.html:
```diff
@@ -39,6 +39,11 @@
         <span class="fa fa-file-text-o"></span> <span translate>REPORT.VIEW_VOUCHER</span>
       </a>
     </li>
+    <li ng-show="row.entity.type_id === 12 || row.entity.type_id === 13 || row.entity.type_id === 14">
```
In the future, it would be cleaner (and probably faster) to just stick an `isStockMovement` to these rows and move this logic into the JS controller.  I'm thinking something like:
```js
const stockMovementTypes = [
  bhConstant.transactionTypes.STOCK_EXIT,
  bhConstant.transactionTypes.STOCK_ENTRY,
  // .. etc
];

rows.forEach(row => {
  row.isStockMovement = stockMovementTypes.includes(row.type_id);
});
```

This isn't a critical change though.

jniles commented on client/src/modules/vouchers/templates/action.cell.html:
```diff
@@ -39,6 +39,11 @@
       </a>
     </li>
+    <li ng-show="row.entity.type_id === 12 || row.entity.type_id === 13 || row.entity.type_id === 14">
+      <a data-method="view-stock-movements" ui-sref="stockMovements({ filters : [{ key : 'voucherReference', value : row.entity.reference, displayValue: row.entity.reference, cacheable:false }, { key : 'period', value : 'allTime', cacheable:false }]})" href>
```
Could you move this and the `data-method="view-voucher"` up in the list?  It seems to be below the reverse record option and I can see users frequently mis-clicking.

![image](https://user-images.githubusercontent.com/896472/113264020-66c27080-92ca-11eb-9baa-732b4ab310c0.png)

jniles reviewed (commented):
This functionality works great!  Good job @lomamech 

The only blocking comment is to move the `<li>` higher in the list.  We don't want users accidentally reversing records when they are supposed to just be looking at the record.

lomamech committed:
improvement(Link Stock Movement from vouchers)\
- Improve action menu, move the link in Up of the list
- Use property isStockMovement in using Constant and vouchers

jniles reviewed (approved):
Looks good to me!

bors r+
